File: server/php/src/Infrastructure/SDKClients/PaymentLinkClient.php
<?php

namespace OnlinePayments\ExampleApp\Infrastructure\SDKClients;

use OnlinePayments\ExampleApp\Application\DTOs\PaymentLink\RequestDto;
use OnlinePayments\ExampleApp\Application\DTOs\PaymentLink\ResponseDto;
use OnlinePayments\ExampleApp\Application\Interfaces\SDKClients\PaymentLinkClientInterface;
use OnlinePayments\ExampleApp\Infrastructure\Mappers\ExceptionMapper;
use OnlinePayments\ExampleApp\Infrastructure\Mappers\PaymentLinkMapper;
use OnlinePayments\Sdk\Merchant\MerchantClient;
use Psr\Log\LoggerInterface;

final readonly class PaymentLinkClient implements PaymentLinkClientInterface
{
    public function __construct(
        private MerchantClient  $merchantClient,
        private LoggerInterface $logger
    )
    {
    }

    /**
     * @throws \Throwable
     */
    public function createPaymentLink(RequestDto $requestDto): ResponseDto
    {
        try {
            $request = PaymentLinkMapper::mapToSdkRequest($requestDto);

            $this->logger->info(
                "The payment link creation has started - Amount: {$request->order->amountOfMoney->amount};\n" .
                "Currency: {$request->order->amountOfMoney->currencyCode}."
            );

            $response = $this->merchantClient->paymentLinks()->createPaymentLink($request);

            $this->logger->info(
                "Generation of the payment link successfully completed - Redirect url: {$response->redirectionUrl}."
            );

            return PaymentLinkMapper::mapFromSdkResponse($response);
        } catch (\Exception $ex) {
            $this->logger->error('Error occurred while creating payment link', [
                'exception' => $ex->getMessage()
            ]);
            throw ExceptionMapper::map($ex);
        }
    }
}

File: server/dotnet/sdk-dotnet-example/Infrastructure/SDKClients/PaymentLinkClient.cs
using Business.DTOs.CreatePaymentLinks;
using Business.Interfaces.SDKClients;
using Infrastructure.Mappers;
using Infrastructure.Mappers.Exceptions;
using OnlinePayments.Sdk;
using OnlinePayments.Sdk.Domain;
using OnlinePayments.Sdk.Merchant;

namespace Infrastructure.SDKClients;

public class PaymentLinkClient(ILogger<PaymentLinkClient> logger, IMerchantClient merchantClient) : IPaymentLinkClient
{
    public async Task<CreatePaymentLinkResponseDto> CreatePaymentLinkAsync(CreatePaymentLinkRequestDto request)
    {
        try
        {
            CreatePaymentLinkRequest requestSdk = PaymentLinkMapper.Map(request);
            logger.LogInformation("Creating payment link request for payment - Amount: {}; Currency: {}.",
                requestSdk.Order.AmountOfMoney.Amount, requestSdk.Order.AmountOfMoney.CurrencyCode);
            PaymentLinkResponse response = await merchantClient.PaymentLinks.CreatePaymentLink(requestSdk);
            logger.LogInformation("Successful payment link - Redirect url: {}.", response.RedirectionUrl);
            CreatePaymentLinkResponseDto responseDto = PaymentLinkMapper.Map(response);

            return responseDto;
        }
        catch (ApiException ex)
        {
            throw ExceptionMapper.Map(ex);
        }
    }
}